File: app/src/main/java/net/bloople/manga/TagChooserFragment.java
package net.bloople.manga;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;

import androidx.fragment.app.DialogFragment;

public class TagChooserFragment extends DialogFragment {
    static TagChooserFragment newInstance(String[] tags) {
        TagChooserFragment fragment = new TagChooserFragment();
        Bundle args = new Bundle();
        args.putStringArray("tags", tags);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Bundle arguments = getArguments();
        final String[] tags = arguments.getStringArray("tags");

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
        builder.setTitle("Search for tag")
                .setItems(tags, (dialog, which) -> {
                    IndexActivity activity = (IndexActivity)getActivity();
                    activity.useTag(tags[which]);
                });
        return builder.create();
    }
}

File: app/src/main/java/net/bloople/manga/TagChooserFragment.kt
package net.bloople.manga

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.content.DialogInterface
import androidx.fragment.app.DialogFragment

class TagChooserFragment : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val arguments = arguments
        val tags = arguments!!.getStringArray("tags")

        val builder = AlertDialog.Builder(activity)
        builder.setTitle("Search for tag")
            .setItems(tags) { _: DialogInterface?, which: Int ->
                val activity = activity as IndexActivity?
                activity!!.useTag(tags!![which])
            }
        return builder.create()
    }

    companion object {
        @JvmStatic
        fun newInstance(tags: Array<String>): TagChooserFragment {
            val fragment = TagChooserFragment()
            val args = Bundle()
            args.putStringArray("tags", tags)
            fragment.arguments = args
            return fragment
        }
    }
}